keps/sig-scheduling/895-pod-topology-spread/README.md

```diff
@@ -520,6 +529,128 @@ simply achieve the semantics of "PodAntiAffinity in zones" via a combination of
 "Even pods spreading in zones" plus "PodAntiAffinity in nodes" which could be an
 extra benefit of this KEP.
 
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  
+  - [x] Feature gate
+    - Feature gate name: EvenPodsSpread
+    - Components depending on the feature gate: kube-scheduler, kube-apiserver
+
+* **Does enabling the feature change any default behavior?**
+
+  No.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+
+  The feature can be disabled in Alpha and Beta versions. In terms of Stable
+  versions, users can choose to opt-out by not setting the
+  `pod.spec.topologySpreadConstraints` field.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  N/A.
+
+* **Are there any tests for feature enablement/disablement?**
+
+  No.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+  Since this feature requires users to opt-in by setting new field in pod spec,
+  it should not impact already running workloads.
+
+* **What specific metrics should inform a rollback?**
+
+  Metric "schedule_attempts_total" remaining at zero when new pods are added.
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+
+  N/A.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?**
+
+  No.
+
+### Monitoring requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+  Operator can query `pod.spec.topologySpreadConstraints` field and identify if
+  this is being set to non-default values.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+
+  N/A.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+  N/A.
+
+* **Are there any missing metrics that would be useful to have to improve
+  observability if this feature?**
+
+  N/A.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+  No.
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+
+  No
+
+* **Will enabling / using this feature result in introducing new API types?**
+
+  No.
+
+* **Will enabling / using this feature result in any new calls to cloud
+  provider?**
+
+  No.
+
+* **Will enabling / using this feature result in increasing size or count
+  of the existing API objects?**
+
+  Since this feature adds a new field to pod's spec, it will increase API size
+  of Pod object depending on how many `topologySpreadConstraints` are specified.
+
+* **Will enabling / using this feature result in increasing time taken by any
+  operations covered by [existing SLIs/SLOs][]?**
+
+  No.
```

Comment:
`framework_extension_point_duration_seconds` and other latency SLI

Reply:
I would say an SLI/SLO can be setup for pod scheduling latency, this feature is replacing default spreading, so it is unlikely to increase pod scheduling latency. ```framework_extension_point_duration_seconds``` will not be a metric that we use to measure an SLO, it is more for debugging purposes.

Reply:
That is true, but there is a difference: default spreading uses only the nodes that pass filters, whereas we use all nodes in Topology spreading. It might we worth saying that some increase in latency might happen, and that it can be monitored with the above metric.